task-maker-async/src/store.rs

```diff
@@ -43,58 +84,1011 @@ pub enum FileReadingOutcome {
 
 #[tarpc::service]
 pub trait Store {
-    /// Used by the client to upload inputs to the DAG. Returns an error if the file set is already
-    /// present. Either way, this leases the corresponding hash for LEASE_LENGTH_SEC.
-    async fn create_input_fileset(hash: InputFileHash) -> Result<FileSetHandle, ()>;
-
-    /// Opens a file for reading. Creates a lease for the input fileset that will prevent it
-    /// from being dropped. If the file is not present and finalized, returns an error immediately.
-    async fn open_input_file(hash: InputFileHash) -> Result<FileSetHandle, ()>;
+    /// Used by the client to upload inputs to the DAG. Returns a writing handle if the input file is
+    /// not present, or a reading handle if it is.
+    async fn create_or_open_input_file(
+        hash: DataIdentificationHash,
+    ) -> Result<FileSetHandle, Error>;
 
     // Creating a computation is not a RPC.
 
-    /// Opens a computed fileset for reading. Creates a lease for the computation data that will prevent it
+    /// Opens a computed file_set for reading. Creates a lease for the computation data that will prevent it
     /// from being dropped. If the computation is not present, waits until it is created.
-    async fn open_computation(hash: ComputationHash) -> FileSetHandle;
+    async fn open_computation(
+        computation: DataIdentificationHash,
+        variant: VariantIdentificationHash,
+    ) -> Result<FileSetHandle, Error>;
 
-    /// Opens a file inside a fileset. Waits for the file to be created if it doesn't exist yet and
-    /// the fileset handle is a reading handle, creates the file otherwise.
+    /// Opens a file inside a file_set. Waits for the file to be created if it doesn't exist yet and
+    /// the file_set handle is a reading handle, creates the file otherwise.
     /// Returns an error if the handle is invalid.
```

Comment:
Well, not _that_ different, I would say (it opens a file for writing, immediately, or for reading, potentially waiting)

Reply:
Yeah, it's not that different (and works very similarly to "fopen"). My comment was more about the wording of that comment. I feed something like this would be easier to understand:

> Opens a file inside a file set. If the handle is read-only, and the file does not exists, it blocks until it is created. If the handle is a writing handle, it creates the file.